api/src/services/authorization.ts

```diff
@@ -472,7 +472,7 @@ export class AuthorizationService {
 		const payloadWithPresets = merge({}, preset, payload);
 
 		const fieldValidationRules = Object.values(this.schema.collections[collection].fields)
-			.map((field) => field.validation)
+			.map((field) => parseFilter(field.validation, this.accountability))
```

Comment:
Should we (somehow) add the third argument, filter context, for nested dynamic variables as well? Explored this exact same solution as well, but wasn't sure how to cleanly abstract out the logic from get-permissions 🤔

https://github.com/directus/directus/blob/726af0c57cf4e3a94c61d828db0dbd811851c4c9/api/src/utils/get-permissions.ts#L195

Just a concern since the docs do currently state [nested user/role variables is available in validation](https://docs.directus.io/configuration/filter-rules/#dynamic-variables).

Reply:
You're very right, will look further into how we could add the filter context.